source/common/runtime/runtime_impl.cc

```diff
@@ -121,6 +121,19 @@ bool SnapshotImpl::featureEnabled(absl::string_view key,
     percent = default_value;
   }
 
+  // When numerator > denominator condition is always evaluates to TRUE
+  // It becomes hard to debug why configuration does not work in case of wrong numerator.
+  // Log debug message that numerator is invalid.
+  uint64_t denominator_value =
+      ProtobufPercentHelper::fractionalPercentDenominatorToInt(percent.denominator());
+  if (percent.numerator() > denominator_value) {
+    ENVOY_LOG(debug,
+              "WARNING runtime key '{}': numerator ({}) > denominator ({}), condition always "
+              "evaluates to true",
+              key, percent.numerator(), denominator_value);
+    NOT_REACHED_GCOVR_EXCL_LINE;
+  }
```

Comment:
I think this check should only occur in debug builds and just throw an exception. Something like:
```
#ifndef NDEBUG
if (percent.numerator() > denominator_value) {
   throw EnvoyException(.....);
}
#endif
```

Reply:
Makes sense.